HealthcareSystem/Backend/Service/ExaminationAndPatientCard/FreeAppointmentSearchService.cs

```diff
@@ -50,6 +55,122 @@ public ICollection<Examination> BasicSearch(BasicAppointmentSearchDTO parameters
             return actuallyAvailableAppointments;
         }
 
+        public ICollection<Examination> GetUnchangedAppointmentsForEmergency(AppointmentSearchWithPrioritiesDTO parameters)
+        {
+            parameters.InitialParameters.EarliestDateTime = GetNewStartTime(parameters.InitialParameters.EarliestDateTime);
+            parameters.InitialParameters.LatestDateTime = parameters.InitialParameters.EarliestDateTime.AddHours(2);
+
+            ICollection<Doctor> allDoctors = _doctorRepository.GetDoctorsBySpecialty(parameters.SpecialtyId);
+            List<Examination> allUnchangedAppointments = new List<Examination>();
+            foreach (Doctor doctor in allDoctors)
+            {
+                parameters.InitialParameters.DoctorJmbg = doctor.Jmbg;
+                allUnchangedAppointments.AddRange(BasicSearch(parameters.InitialParameters));
+            }
+            allUnchangedAppointments.ForEach(e => e.ExaminationStatus = ExaminationStatus.AVAILABLE);  
+
+            return allUnchangedAppointments;
+        }
+
+        public ICollection<Examination> GetShiftedAndSortedAppoinmentsForEmergency(AppointmentSearchWithPrioritiesDTO parameters)
+        {
+            parameters.InitialParameters.EarliestDateTime = GetNewStartTime(parameters.InitialParameters.EarliestDateTime);
+            parameters.InitialParameters.LatestDateTime = parameters.InitialParameters.EarliestDateTime.AddHours(2);
+
+            List<Examination> unavailableAppointments = GetUnavailableAppointments(parameters.InitialParameters);
+            List<Examination> adequateAppoinments = GetOnlyAdequateAppointments(unavailableAppointments, parameters);
+
+            ICollection<Examination> shiftedAppointments = new List<Examination>();
+            foreach (Examination examination in unavailableAppointments)
+            {
+                List<Examination> availableAppointments = GetShiftedAppointmentForEmergency(parameters.InitialParameters, examination);
+                availableAppointments = availableAppointments.OrderBy(e => e.DateAndTime).ToList();
+                foreach (Examination appointment in availableAppointments)
+                {
+                    if (IsShiftedAppoinmentAvailable(appointment, shiftedAppointments))
+                    {
+                        shiftedAppointments.Add(appointment);
+                        break;
+                    }
+                }
+            }
+
+            return shiftedAppointments;
+        }
```

Comment:
If there isn't any free appointment (check AppointmentController for this), then we start appointment analysis. Firstly, we get unavailableAppointments for the wanted period and those appointments are filtered since e.g. if the appointment is by a doctor that doesn't have our specialization there isn't a need for its rescheduling since we won't be able to schedule it anyway. For each appointment, a new schedule is being searched and we add a first available appointment to the final list of shiftedAppointments.